pkg/apis/psmdb/v1/psmdb_types.go

```diff
@@ -930,6 +931,19 @@ type BackupTaskSpec struct {
 	Type defs.BackupType `json:"type,omitempty"`
 }
 
+type BackupTaskSpecRetentionType string
+
+const (
+	BackupTaskSpecRetentionTypeCount = "count"
+)
+
+type BackupTaskSpecRetention struct {
+	Count int `json:"count,omitempty"`
+	// +kubebuilder:validation:Enum={count}
+	Type              string `json:"type,omitempty"`
+	DeleteFromStorage *bool  `json:"deleteFromStorage,omitempty"`
+}
+
```

Comment:
I believe that we could have a 1-1 mapping from `keep` to `retention` without using keep at all in our codebase in a similar way we implemented it in PXC. This way, when we will officially depricate the `keep` option, we will not have to change the actual logic. 

Check here: https://github.com/percona/percona-xtradb-cluster-operator/commit/1553be6da698446eedf067592701e12f7f15a70e#diff-012d1e07c013a6332356a3ab856e8b0db6825aa9021d56c68ea5114faee14705